Description: Reproducible builds, the easy way
Author: mirabilos <tg@debian.org>
Forwarded: https://github.com/musescore/sftools/pull/20

--- a/sfont.cpp
+++ b/sfont.cpp
@@ -1199,8 +1199,7 @@ int SoundFont::writeCompressedSample(Sam
       vorbis_comment_init(&vc);
       vorbis_analysis_init(&vd, &vi);
       vorbis_block_init(&vd, &vb);
-      srand(time(NULL));
-      ogg_stream_init(&os, rand());
+      ogg_stream_init(&os, 0);
 
       ogg_packet header;
       ogg_packet header_comm;
